src/components/structures/LoggedInView.js

```diff
@@ -80,10 +77,19 @@ export default React.createClass({
         this._scrollStateMap = {};
 
         document.addEventListener('keydown', this._onKeyDown);
+
+        this._sessionStore = sessionStore;
+        this._removeSSListener = this._sessionStore.addListener(
+            this._setStateFromSessionStore,
+        ).remove;
+        this._setStateFromSessionStore();
     },
 
     componentWillUnmount: function() {
         document.removeEventListener('keydown', this._onKeyDown);
+        if (this._removeSSListener) {
+            this._removeSSListener();
```

Comment:
does this actually work right? I can imagine it getting confused about what `this` is within the `remove` function.

If it works I guess it's fair enough, though personally I would just store the token and call `remove` on it here explicitly.